base_layer/core/src/base_node/rpc/http/client.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright 2025 The Tari Project
+// SPDX-License-Identifier: BSD-3-Clause
+
+use reqwest::Url;
+
+use crate::{
+    base_node::rpc::{
+        models::TipInfoResponse,
+        BaseNodeWalletQueryServiceClient,
+        BaseNodeWalletQueryServiceClientError,
+    },
+    blocks::BlockHeader,
+};
+
+/// HTTP client for the Base Node wallet query service.
+#[derive(Clone)]
```

Comment:
I think it makes more sense to put this in it's own crate

Reply:
Good point thanks! Moved to a separate crate